File: android/src/main/java/au/com/crazybean/mobilex/kurir/dependency/signup/SignupModule.kt
package au.com.crazybean.mobilex.kurir.dependency.signup

import au.com.crazybean.mobilex.kurir.modules.auth.signup.SignupDelegate
import au.com.crazybean.mobilex.kurir.modules.auth.signup.SignupViewModel
import au.com.crazybean.mobilex.kurir.modules.auth.signup.SignupView
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module

val signupModule = module {
    // Delegate
    factory { (view: SignupView?) ->
        SignupDelegate(view, get())
    }

    // ViewModel
    viewModel {
        SignupViewModel(get(), get())
    }
}

File: ios/Kurir/modules/auth/signup/impl/SignupModule.swift
//
//  SignupModule.swift
//  Kurir
//
//  Created by Loren on 1/9/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Foundation
import Mobilex

class SignupModule: Module {
    override func inject() {
        // ViewModel
        factory(SignupViewModel.self) { r in
            SignupViewModel(r.resolve(UsersRepository.self)!)
        }
        
        // Delegate
        factory(SignupDelegate.self) { (r, view: SignupView) -> SignupDelegate in
            SignupDelegate(view: view, viewModel: r.resolve(SignupViewModel.self)!)
        }
    }
}
